File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/PartTime.swift


public class PartTime: Employee {
    
    private var _hourlyRate : Int
    public var hourlyRate : Int {
        get {
            return _hourlyRate
        }
        
        set {
            _hourlyRate = newValue
        }
    }
    
    private var _numberHoursWorked : Int
    public var numberHoursWorked : Int {
        get {
            return _numberHoursWorked
        }
        
        set {
            _numberHoursWorked = newValue
        }
    }
    
    override init() {
        _hourlyRate = 0
        _numberHoursWorked = 0
        super.init();
    }
    
    init(ppName: String, ppAge: Int, pHourlyRate: Int, pNumberHoursWorked: Int) {
        _hourlyRate = pHourlyRate
        _numberHoursWorked = pNumberHoursWorked
        super.init(ppName, ppAge)
    }
    
    init(ppName: String, ppAge: Int, pHourlyRate: Int, pNumberHoursWorked: Int, ppV: Vehicle) {
        _hourlyRate = pHourlyRate
        _numberHoursWorked = pNumberHoursWorked
        super.init(ppName, ppAge, ppV)
    }
    
    override func getTypeOfEmployee() -> String {
        return "Part Time"
    }
    
    override func printMyData() -> String {
        return "\(super.printMyData())\n  -Rate: \(hourlyRate)\n  -Hours Worked: \(numberHoursWorked)"
    }
}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/employee/PartTime.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.employee;

import c0711561.f2017.mad3464.finalproject.vehicle.Vehicle;
import java.math.BigDecimal;

/**
 *
 * @author macstudent
 */
public class PartTime extends Employee {
    
    private final double rate;
    private final int hoursWorked;

    public PartTime(String name, int age, double rate, int hoursWorked) {
        super(name, age);
        this.rate = rate;
        this.hoursWorked = hoursWorked;
    }
    
    public PartTime(String name, int age, double rate, int hoursWorked, Vehicle vehicle) {
        super(name, age, vehicle);
        this.rate = rate;
        this.hoursWorked = hoursWorked;
    }

    public double getRate() {
        return rate;
    }

    public int getHoursWorked() {
        return hoursWorked;
    }

    @Override
    public BigDecimal calcEarnings() {
        double earnings = rate * hoursWorked;
        return new BigDecimal(earnings);
    }

    @Override
    public String getTypeOfEmployee() {
        return "Part Time";
    }

    @Override
    public String printMyData() {
        return super.printMyData() 
                + "\n -Rate: " + rate
                + "\n -Hours Worked: " + hoursWorked;
    }
}
